docs/how-to/HA.md

```diff
@@ -1,22 +1,67 @@
 
-- Gossip Protocol: This protocol facilitates PMM servers to discover and share information about their states with each other. It is used for managing the PMM server list and failure detection.
-- Raft Protocol: This is a consensus algorithm that allows PMM servers to agree on a leader and ensures that logs are replicated among all machines.
+Once the connection is restored, the cached data is transferred to the PMM Server, ensuring no data loss during the restart process.
+
+This option is suitable for scenarios where the primary concern is the ability to investigate potential issues later. However, it's important to note that this approach is limited by the underlying physical infrastructure. If the failure stems from a hardware issue, automatic recovery might be challenging.
+
+### 2. Leverage Kubernetes for enhanced isolation
+
+If you are running PMM in a Kubernetes (K8s) environment, PMM offers a Helm chart that facilitates running PMM with enhanced isolation. In this setup, even if the physical infrastructure encounters a problem, K8s automatically handles failover, migrating the PMM instance to a healthy node.
+
+While restarts within K8s can take up to several minutes (depending on your infrastructure configuration), PMM's data caching ensures that information is preserved during this transition. Alerts will still be triggered to keep you informed about any issues that started during PMM's restart and continue after PMM is back.
+
+### 3. Fully-clustered PMM in Kubernetes (coming Q3/2024)
+
+If you have a large deployment with numerous instances and distributed locations, you might find that a fully clustered PMM setup in Kubernetes is better suited to your needs. We are actively developing this solution, which is slated for release in Q3/2024, to cater specifically to users managing extensive and complex monitoring environments.
+
+This option will provide a comprehensive HA solution, including clustered database setups (ClickHouse, VictoriaMetrics, and PostgreSQL). In this setup, multiple PMM instances will be configured, with one being the leader and the others as followers.
+
+Leader election will be managed using the Raft consensus algorithm, ensuring a smooth transition of the leader role if the current leader fails. The architecture will consist of:
+
+- Multiple PMM instances for redundancy
+- Clustered PostgreSQL for storing metadata and configuration data
+- Clustered ClickHouse for storing query performance metrics (Query Analytics)
+- Clustered VictoriaMetrics for storing operational metrics from monitored databases and hosts
+- HAProxy for managing and directing network traffic to the current leader PMM instance
+
+This fully clustered PMM setup in Kubernetes will provide the highest level of availability and reliability for large-scale, mission-critical monitoring environments.
 
 ## Prerequisites
```

Comment:
everything after this line is like 4th option

Reply:
So I think we should write some explanation like `if all other options doesn't work for you, you can use this manual written below`